flytekit/core/array_node.py

```diff
@@ -0,0 +1,214 @@
+import math
+from typing import Any, List, Optional, Set, Tuple, Union
+
+from flytekit.core import interface as flyte_interface
+from flytekit.core.context_manager import ExecutionState, FlyteContext
+from flytekit.core.interface import transform_interface_to_list_interface, transform_interface_to_typed_interface
+from flytekit.core.launch_plan import LaunchPlan
+from flytekit.core.node import Node
+from flytekit.core.promise import (
+    Promise,
+    VoidPromise,
+    flyte_entity_call_handler,
+    translate_inputs_to_literals,
+)
+from flytekit.core.task import TaskMetadata
+from flytekit.loggers import logger
+from flytekit.models import literals as _literal_models
+from flytekit.models.core import workflow as _workflow_model
+from flytekit.models.literals import Literal, LiteralCollection, Scalar
+
+
+class ArrayNode(object):
+    def __init__(
+        self,
+        target: LaunchPlan,
+        concurrency: Optional[int] = None,
+        min_successes: Optional[int] = None,
+        min_success_ratio: Optional[float] = None,
+        bound_inputs: Optional[Set[str]] = None,
+        execution_version: Optional[int] = None,
+        metadata: Optional[Union[_workflow_model.NodeMetadata, TaskMetadata]] = None,
```

Comment:
I forgot to clean this up when removing code. When I initially implemented this, this new ArrayNode class supported PythonTasks as well. Since we still plan to add this support as a follow up, would it be fine to leave this as is?

Reply:
fair. Leave it as is.